src/SmartComponents/DriftPage/DriftTable/DriftTable.js

```diff
@@ -91,7 +110,11 @@ export class DriftTable extends Component {
         }
         /*eslint-enable camelcase*/
 
-        return baselines.concat(fullHistoricalSystemList);
+        masterList =
+            this.referenceId
+                ? this.moveSystem(baselines.concat(fullHistoricalSystemList))
+                : baselines.concat(fullHistoricalSystemList);
```

Comment:
This performs a concat in both cases so I would opt to simplify this and move that up a line something like:

```
masterList = baselines.concat(fullHistoricalSystemLIst);

if (this.referenceId) {
  masterList = this.moveSystem(masterList);
}
```